### PR TITLE
Add commands to retrieve details about artifacts deployed in MI instances

### PR DESCRIPTION
**Pull request by SanojPunchihewa**
## Purpose
Update MI CLI tool and merge with APICTL
wso2/micro-integrator#1999

## Goals
- Add a new get command under mi alias to retrieve details about artifacts deployed in MI instances

## Approach
- Created a new package called get inside the mi package
- Add the necessary sub-commands inside that get package
- Created a new root level package called mi
- Add all the implementation related to retrieving and printing of artifacts inside a sub package called impl inside that mi package

## User stories
Following are the new commands introduced by this PR to retrieve details about artifacts deployed in MI instances.

| Command                           | Details                                                         |
|-----------------------------------|-----------------------------------------------------------------|
| apictl mi get apis                | Get information about apis                                      |
| apictl mi get composite-apps      | Get information about composite apps                            |
| apictl mi get connectors          | Get information about connectors                                |
| apictl mi get data-services       | Get information about data services                             |
| apictl mi get endpoints           | Get information about endpoints                                 |
| apictl mi get inbound-endpoints   | Get information about inbound endpoints                         |
| apictl mi get local-entries       | Get information about local entries                             |
| apictl mi get log-levels          | Get information about a Logger configured in a Micro Integrator |
| apictl mi get logs                | List all the available log files                                |
| apictl mi get message-processors  | Get information about message processors                        |
| apictl mi get message-stores      | Get information about message stores                            |
| apictl mi get proxy-services      | Get information about proxy services                            |
| apictl mi get sequences           | Get information about sequences                                 |
| apictl mi get tasks               | Get information about tasks                                     |
| apictl mi get templates           | Get information about templates                                 |
| apictl mi get transaction-counts  | Retrieve transaction count                                      |
| apictl mi get transaction-reports | Generate transaction count summary report                       |
| apictl mi get users               | Get information about users                                     |

## Test environment
Ubuntu 20.04.1 LTS
Go version go1.15 linux/amd64

### CODE REVIEW

File: import-export-cli/cmd/mi/get/apis.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	if len(args) == 1 {
+		var IntegrationAPIName = args[0]
+		executeShowIntegrationAPI(IntegrationAPIName)
+	} else {
+		executeListIntegrationAPIs()
+	}
+}
+
+func executeListIntegrationAPIs() {
+
```
[the remark this answers is not in the document]
Comment (SanojPunchihewa) [reply import-export-cli/cmd/mi/get/apis.go]: Fixed with c5e4b62

File: import-export-cli/cmd/mi/get/compositeApps.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	if len(args) == 1 {
+		var appName = args[0]
+		executeShowCarbonApp(appName)
+	} else {
+		executeListCarbonApps()
+	}
+}
+
+func executeListCarbonApps() {
+
```
[the remark this answers is not in the document]
Comment (SanojPunchihewa) [reply import-export-cli/cmd/mi/get/compositeApps.go]: Fixed with c5e4b62

File: import-export-cli/cmd/mi/get/connectors.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+func handleGetConnectorCmdArguments(args []string) {
+	printGetCmdVerboseLogForArtifact(getConnectorCmdLiteral)
+	credentials.HandleMissingCredentials(getConnectorCmdEnvironment)
+	executeListConnectors()
+}
+
+func executeListConnectors() {
+
```
[the remark this answers is not in the document]
Comment (SanojPunchihewa) [reply import-export-cli/cmd/mi/get/connectors.go]: Fixed with c5e4b62

File: import-export-cli/cmd/mi/get/dataservices.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	if len(args) == 1 {
+		var dataServiceName = args[0]
+		executeShowDataService(dataServiceName)
+	} else {
+		executeListDataServices()
+	}
+}
+
+func executeListDataServices() {
+
```
[the remark this answers is not in the document]
Comment (SanojPunchihewa) [reply import-export-cli/cmd/mi/get/dataservices.go]: Fixed with c5e4b62

File: import-export-cli/mi/impl/getIntegrationApis.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+		"Stats - {{.Stats}}\n" +
+		"Tracing - {{.Tracing}}\n" +
+		"Resources :\n" +
+		"URL\tMETHOD\n" +
+		"{{range .Resources}}{{.Url}}\t{{.Methods}}\n{{end}}"
+)
+
+// GetIntegrationAPIList returns a list of apis deployed in the micro integrator in a given environment
+func GetIntegrationAPIList(env string) (*artifactutils.IntegrationAPIList, error) {
+
```
[the remark this answers is not in the document]
Comment (SanojPunchihewa) [reply import-export-cli/mi/impl/getIntegrationApis.go]: Fixed with c5e4b62

File: import-export-cli/mi/impl/getIntegrationApis.go [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+		return nil, err
+	}
+	return resp.(*artifactutils.IntegrationAPIList), nil
+}
+
+// PrintIntegrationAPIList print a list of apis according to the given format
+func PrintIntegrationAPIList(apiList *artifactutils.IntegrationAPIList, format string) {
+
+	if apiList.Count > 0 {
+
```
[the remark this answers is not in the document]
Comment (SanojPunchihewa) [reply import-export-cli/mi/impl/getIntegrationApis.go]: Fixed with c5e4b62

File: import-export-cli/cmd/mi/get/apis.go [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+	Run: func(cmd *cobra.Command, args []string) {
+		handleGetIntegrationAPICmdArguments(args)
+	},
+}
+
+func init() {
+	GetCmd.AddCommand(getIntegrationAPICmd)
+	getIntegrationAPICmd.Flags().StringVarP(&getIntegrationAPICmdEnvironment, "environment", "e",
+		"", "Environment to be searched")
+	getIntegrationAPICmd.Flags().StringVarP(&getIntegrationAPICmdFormat, "format", "", "", generateFormatFlagUsage(artifactAPIs))
```
[the remark this answers is not in the document]
Comment (SanojPunchihewa) [reply import-export-cli/cmd/mi/get/apis.go]: Refactored these methods with c5e4b62

File: import-export-cli/cmd/mi/get/apis.go [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+	Short:   generateGetCmdShortDescForArtifact(artifactAPIs),
+	Long:    generateGetCmdLongDescForArtifact(artifactAPIs, "api-name"),
+	Example: generateGetCmdExamplesForArtifact(artifactAPIs, getTrimmedCmdLiteral(getIntegrationAPICmdLiteral), "SampleIntegrationAPI"),
+	Args:    cobra.MaximumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		handleGetIntegrationAPICmdArguments(args)
+	},
+}
+
+func init() {
```
[the remark this answers is not in the document]
Comment (SanojPunchihewa) [reply import-export-cli/cmd/mi/get/apis.go:44]: The current implementation of the Rest API will return all the APIs deployed in the micro integrator

**Review by wasuradananjith — APPROVED** [review]

**Review by 1akshitha — APPROVED** [review]

**Pull request merged**